### PR TITLE
feat(api) : activity stream, nextRef & indentifier configured into orchestrator

### PR DESCRIPTION
**Pull request by soumyadip007**
## Closes

- Identifier added for spa props
- Activity Stream configured
- NextRef added for spa versioning

## Explain the feature

- Identifier added into application collection 
- SPA listing response changed (identifier added)
- Identifier added to Consume Events config (operator based on identifier)
- Changes in Activity stream collection (id removed)
- Activity stream for creating a new property, creating new env, creating API key, uploading spa

## Does this PR introduce a breaking change

- No

### CODE REVIEW

File: packages/api/controllers/application.js [hunk 1]
```diff
@@ -96,16 +100,18 @@
       });
 
       if (response) {
-        const application = await Application.findOne({ propertyName, name, path: appPath, env });
+        const application = await Application.findOne({ propertyName, identifier, path: appPath, env });
         try {
           if (application) {
-            await Application.updateOne({ propertyName, name, path: appPath, env }, { ref });
+            await Application.updateOne({ propertyName, identifier, path: appPath, env }, { name, nextRef });
```
[the remark this answers is not in the document]
Comment (soumyadip007) [reply packages/api/controllers/application.js:106]: This is a false positive. We are using this from the start, and using the pre-santitized data.

File: packages/api/controllers/application.js [hunk 2]
```diff
@@ -96,16 +100,18 @@
       });
 
       if (response) {
-        const application = await Application.findOne({ propertyName, name, path: appPath, env });
+        const application = await Application.findOne({ propertyName, identifier, path: appPath, env });
```
[the remark this answers is not in the document]
Comment (soumyadip007) [reply packages/api/controllers/application.js:103]: This is a false positive. We are using this from the start, and using the pre-santitized data.

File: packages/api/controllers/application.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (soumyadip007) [reply packages/api/controllers/application.js:106]: This is a false positive. We are using this from the start, and using the pre-santitized data.

**Review by arkaprovob — APPROVED** [review]

File: packages/api/controllers/application.js [hunk 3]
```diff
@@ -276,3 +293,19 @@ function validateProperties(request, next) {
...
+    .replace(/%[0-9a-zA-Z]{2}/g, '-')
+    /* Replace any special characters with `-` */
+    .replace(/[\ \-\/\:\@\[\]\`\{\~\.]+/g, '-')
+    /* Special characters are replaced by an underscore */
+    .replace(/[\|!@#$%^&*;"<>\(\)\+,]/g, '_')
+    /* Remove any starting or ending `-` */
+    .replace(/^-+|-+$/g, '')
+    /* Removing multiple consecutive `-`s */
+    .replace(/--+/g, '-');
+}
```
Comment (hybridx) [inline packages/api/controllers/application.js:311]: @soumyadip007 do test this function before using it

**Review by hybridx — APPROVED** [review]

**Comment by hybridx** [comment]
@soumyadip007 I would recommend to add swagger with this PR

File: packages/api/controllers/application.js (hunk 3 above)
Comment (soumyadip007) [reply packages/api/controllers/application.js:311]: Yes @hybridx . I've tested it.

**Pull request merged**